experimenter/experimenter.py

```diff
@@ -84,80 +85,74 @@ def __init__(self, datasets_dir: str, volumes_dir: str, input_problem_directory=
             self.num_pipelines = len(self.generated_pipelines["classification"])
             self.output_automl_pipelines_to_mongodb()
 
-    def _add_initial_steps(self, pipeline_description: Pipeline, step_counter: int) -> Tuple[int, int, int]:
+    def _add_initial_steps(self, pipeline_description: EZPipeline) -> None:
```

Comment:
Should we put this in the EZPipeline class as a method?  I'm fine either way, just wondering where this function would best live.

Reply:
It doesn't have to be in this PR either (or may be better to just leave it as a function that modifies an object - as it is currently).

This PR looks good and abstracts away a lot.  I really like the way these changes are going!

Reply:
Thanks Orion, those might be a nice addition. I think that would be good to add in a coming PR. Along the lines of what you're saying, we could move `add_pipeline_step` into the `EZPipeline` class. The `create_pipeline_step` helper is decoupled from a specific instance of a pipeline, but we could potentially get rid of it altogether by exposing the current step of an `EZPipeline` as an attribute. That way if you need to add extra arguments or hyperparameters to a pipeline's step, you could just do something like:

```python
pipeline = EZPipeline()
pipeline.add_input(name='inputs')

# It'd be important to not call the `EZPipeline` version of the
# `add_pipeline_step` helper that's used here `add_step`, since
# the d3m Pipeline class already uses that name. 
pipeline.add_basic_step(
    'd3m.primitives.data_transformation.dataset_to_dataframe.Common',
    'inputs.0'
)

# `pipeline.curr_step` could then expose the pipeline's current
# `PrimitiveStep` object.
pipeline.curr_step.add_hyperparameter(...)
```